File: Models/Billing.cs
using System;
using System.Collections.Generic;
using WebAPI_NRW.Models.Database;

namespace WebAPI_NRW.Models;

public partial class Billing
{
    public int Id { get; set; }

    public int SanLuongTieuThu { get; set; }

    public string MaDoiTuong { get; set; } = null!;

    public int Ky { get; set; }

    public int Nam { get; set; }

    public int? Dot { get; set; }

    public DateOnly? TuNgay { get; set; }

    public DateOnly? DenNgay { get; set; }

    public string? GhiChu { get; set; }

    public DateTime NgayTao { get; set; }

    public DateTime? NgayCapNhat { get; set; }

    public int NguoiTao { get; set; }

    public int? NguoiCapNhat { get; set; }

    public virtual NguoiDung? NguoiCapNhatNavigation { get; set; }

    public virtual NguoiDung NguoiTaoNavigation { get; set; } = null!;
}

File: src/types/he-thong-billing/billing.ts
/* ========== REQUEST (CLIENT --> API) ========== */
// Thêm dữ liệu
export interface AddBillingRequest {
  sanLuongTieuThu: number;
  maDoiTuong: string;
  ky: number;
  nam: number;
  dot?: number;
  tuNgay?: string;
  denNgay?: string;
  ghiChu?: string;
  ngayTao: string;
  nguoiTao: string;   // string, giống BE
}

// Cập nhật dữ liệu
export interface UpdateBillingRequest {
  sanLuongTieuThu: number;
  maDoiTuong: string;
  ky: number;
  nam: number;
  dot?: number;
  tuNgay?: string;
  denNgay?: string;
  ghiChu?: string;
  ngayCapNhat?: string;
  nguoiCapNhat?: string;  // string, giống BE
}

/* ========== RESPONSE (API --> CLIENT) ========== */
export interface BillingResponse {
  id: number;
  sanLuongTieuThu: number;
  maDoiTuong: string;
  ky: number;
  nam: number;
  dot?: number;
  tuNgay?: string;
  denNgay?: string;
  ghiChu?: string;
  ngayTao: string;
  ngayCapNhat?: string;
  nguoiTao?: string;
  nguoiCapNhat?: string;
}
